test_infra/test_infra/base_stack.py

```diff
@@ -0,0 +1,129 @@
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import aws_glue as glue
+from aws_cdk import aws_iam as iam
+from aws_cdk import aws_kms as kms
+from aws_cdk import aws_logs as logs
+from aws_cdk import aws_s3 as s3
+from aws_cdk import core as cdk
+
+
+class BaseStack(cdk.Stack):  # type: ignore
+    def __init__(self, scope: cdk.Construct, construct_id: str, **kwargs: str) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        self.vpc = ec2.Vpc(
+            self,
+            "aws-data-wrangler-vpc",
+            cidr="11.19.224.0/19",
+            enable_dns_hostnames=True,
+            enable_dns_support=True,
+        )
+        self.key = kms.Key(
+            self,
+            id="aws-data-wrangler-key",
+            description="Aws Data Wrangler Test Key.",
+            policy=iam.PolicyDocument(
+                statements=[
+                    iam.PolicyStatement(
+                        sid="Enable IAM User Permissions",
+                        effect=iam.Effect.ALLOW,
+                        actions=["kms:*"],
+                        principals=[iam.AccountRootPrincipal()],
+                        resources=["*"],
+                    )
```

Comment:
Isn't this piece missing? https://github.com/awslabs/aws-data-wrangler/blob/b13fcd8d169feb3219b4b4fff025dc6089cfe03b/cloudformation/base.yaml#L175

Reply:
Hmm, I don't think so - the above statement already allows the root principal (which means effectively any inhabitant of the account) all KMS actions.

Reply:
No, I know it's confusing but I don't think that is what it actually means. “root” in a KMS policy only means that access to a KMS key can be granted to anyone **if they have the IAM policy that enables them to do so**. In other words the “root” kms policy alone is not enough to use the key, they still need an IAM role that provides access to that specific key: https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html#key-policy-default

Specifically from the docs:
 
Giving the AWS account full access to the CMK does this; it enables you to use IAM policies to give IAM users and roles in the account access to the CMK. It does not by itself give any IAM users or roles access to the CMK, but it enables you to use IAM policies to do so. For more information, see [Managing access to AWS KMS CMKs](https://docs.aws.amazon.com/kms/latest/developerguide/control-access-overview.html#managing-access).